### PR TITLE
Ci fixes

### PR DESCRIPTION
**Pull request by AnthonyMichaelTDM**
make the cli CI routines create a schema.rs that is usable for testing, bypassing the need for diesel in testing and allowing CI to pass

also fixes a few clippy lints

### CODE REVIEW

**Comment by AnthonyMichaelTDM** [comment]
I mark it as ready for review once I've figured this out

**Comment by AnthonyMichaelTDM** [comment]
okay I'm stumped now, for reference the Actix + Postgres is the one that should definitely be working, but it isn't

**Comment by AnthonyMichaelTDM** [comment]
Any thoughts?

**Comment by Wulf** [comment]
hey @AnthonyMichaelTDM I'll try to take a look next weekend but no promises :sweat_smile:

**Comment by AnthonyMichaelTDM** [comment]
Ty lol

**Comment by Wulf** [comment]
hey buddy, you just need to change `echo` to `printf`:)

The rest looks great :raised_hands:

**Comment by Wulf** [comment]
I'm talking about the `echo` you use to create the schema file!

**Comment by AnthonyMichaelTDM** [comment]
something is still iffy, I'll handle it later my friend got into a car crash

**Comment by AnthonyMichaelTDM** [comment]
Looks like some are failing bc of of a error in how the CI is written, other failures seem to be caused by actual bugs though so I'll work on rooting those out when I get home

**Comment by AnthonyMichaelTDM** [comment]
Going to supersede this with a new PR if/when #197 is merged, going to add the clippy fixes and optimizations to CI.yml to that PR as well

**Comment by AnthonyMichaelTDM** [comment]
when I come back to this, I'll probably remove some CI checks,

also, a lot of the "no plugins" checks are failing because of actual bugs, not just issues in the CI like the other checks

**Comment by AnthonyMichaelTDM** [comment]
That bundle of commits is from me synchronizing this branch with Wulf:main to fix some of the issues that were popping up

**Comment by AnthonyMichaelTDM** [comment]
I think I finally figured it out @Wulf 

there were 3 issues:
- the test-project wasn't using the local create-rust-app, it was using the one published to crates.io (meaning it also experienced the bugs in that version)
- the cache was caching build artifacts from the test-project, stopping the build step from actually building a test-project, leaving it an empty folder with a .cargo directory
- the working directory of commands after cd ... was still project root

**Comment by AnthonyMichaelTDM** [comment]
now it should just be a matter of chasing down actual bugs ^.^

**Comment by AnthonyMichaelTDM** [comment]
now, poem builds are failing due to a bug in create-rust-app/src/util/poem_utils.rs

and sqlite builds are failing due to an error on my end regarding the schema.rs file added in the CI

**Comment by AnthonyMichaelTDM** [comment]
found the cause of the utoipa issue 

https://github.com/Wulf/create-rust-app/blob/fd10180d9d071d84e43fb9dad39764a7ec312aa7/create-rust-app_cli/src/plugins/utoipa.rs#L81-L106

this is adding the dependencies to main a second time, hence the double import issues

**Comment by AnthonyMichaelTDM** [comment]
sql bug fixed a while ago

poem bug fix in progress (should be next commit)

utoipa bug fix in progress

**Comment by AnthonyMichaelTDM** [comment]
poem issue fixed

**Comment by AnthonyMichaelTDM** [comment]
the remaining errors are referencing bugs I already fixed, maybe it's pulling the version from crate.io? going to remove the cache to see if that fixes it

**Comment by AnthonyMichaelTDM** [comment]
Ohhhh.... the issue is in dsync itself, as in the library

and is caused because dsync depends on create-rust-app

File: create-rust-app/src/util/poem_utils.rs [hunk 1]
```diff
@@ -127,15 +134,14 @@ pub async fn render_views(uri: &Uri) -> impl IntoResponse {
...
     let mut content = content;
     #[cfg(debug_assertions)]
     {
-        let uri = Uri::from_str(req.connection_info().host());
-        let hostname = match &uri {
-            Ok(uri) => uri.host().unwrap_or("localhost"),
-            Err(_) => "localhost",
-        };
+        let hostname = _uri.host().unwrap_or("localhost");
         let inject: &str = &format!(
```
Comment (AnthonyMichaelTDM) [inline create-rust-app/src/util/poem_utils.rs:145]: This is the only someone significant change that isn't directly related to formatting or the CI checks is.

**Comment by AnthonyMichaelTDM** [comment]
as that is the only change that *could* change behavior, and the previous behavior prevented compilation, and I'm 99.9% sure the new behavior achieves the intent of the old behavior, I'm going to merge this

**Comment by AnthonyMichaelTDM** [comment]
the remaining things causing CI failures will be fixed when these changes are published to crates.io

**Comment by AnthonyMichaelTDM** [comment]
> the remaining things causing CI failures will be fixed when these changes are published to crates.io

found the real culprit of these bugs, fixed in #226

**Pull request merged**